PostgreSQL refactor: handle all data returned by native functions in DB_pgsql::simpleQuery()

DanCld opened the pull request:
This refactor was inspired by the need to support a native PostgreSQL feature, that is handling data returned by DML queries such as INSERT or UPDATE.

In practical terms, PostgreSQL supports queries like "INSERT INTO ... VALUES ... RETURNING..." that can be used to easily retrieve the last insertion id, although the feature is far more extensive, as per docs: https://www.postgresql.org/docs/current/dml-returning.html

This refactor did away with the strong differentiation between the "read" and "write" code paths, while only relying on the response received from the PostgreSQL backend. This also allows for better error handling regarding errors returned during the actual query execution.

DanCld committed:
pgsql query refactor: always pass on the returned result

DanCld committed:
comments and spacing

schengawegga commented on DB/pgsql.php:
```diff
@@ -353,32 +353,31 @@ function simpleQuery($query)
+            // this query has returned data
+            $this->row[$this->_resultId($result)] = 0; // reset the row counter.
             $numrows = $this->numRows($result);
             if (is_object($numrows)) {
```
I think there is a problem with this if-statement.
According to the [documentation](https://www.php.net/manual/en/function.pg-num-rows.php), the function pg_num_rows() returns a '-1' value on error.
In the method numRows() here, it checks for a 'null' response, to identify an error.
I think this is not correct in numRows().
Can you check this, please?
I don´t have any PG instance to check this on my own.
Maybe the failure handling in this method has never worked well...... ;-)

The other changes looks good to me.

DanCld committed:
fix for numRows() method

pg_num_rows() only returns int values, so the null result comparison never achieved the proper intent.
